log/tracing/middleware.js

```diff
@@ -0,0 +1,12 @@
+const uuid = require('uuid')
+
+const { REQUEST_ID_HEADER, ROOT_REQUEST_ID_HEADER } = require('./headers')
+
+function requestTracingMiddleware (req, res, next) {
+  const id = uuid()
+  req.headers[REQUEST_ID_HEADER] = id
+  req.headers[ROOT_REQUEST_ID_HEADER] = id
```

Comment:
shouldn't these check if they are already present?
just because it is node doesn't mean it is the root request?

Reply:
Fair point 👍